src/tactic/lint/misc.lean

```diff
@@ -319,3 +320,31 @@ meta def linter.check_univs : linter :=
 Note: if the linter flags an automatically generated declaration `xyz._proof_i`, it means that
 the universe problem is with `xyz` itself (even if the linter doesn't flag `xyz`)",
   is_fast := tt }
+
+/-!
+## Linter for syntactic tautologies
+-/
+
+/--
+Checks whether a lemma is a declaration that `∀ a b ... z,e₁ = e₂` where `e₁` and `e₂` are equal
+exprs, we call declarations of this form syntactic tautologies.
+Such lemmas are (mostly) useless and sometimes introduced unintentionally when proving basic facts
+with rfl when elaboration results in a different term than the user intended.
+-/
+meta def syn_taut (d : declaration) : tactic (option string) := do
+  let l := d.type.pi_codomain,
+  (do (el, er) ← expr.is_eq l,
+    -- possible to use alpha equivalence here but seems unlikely to make a difference
+    guard (el = er),
+    return $ some "LHS equals RHS syntactically") <|>
+  return none
+
+/-- A linter for checking that declarations aren't syntactic tautologies. -/
+@[linter]
+meta def linter.syn_taut : linter :=
+{ test := syn_taut,
+  auto_decls := ff, -- many false positives with this enabled
+  no_errors_found :=
+    "No declarations are syntactic tautologies.",
+  errors_found := "THE FOLLOWING DECLARATIONS ARE SYNTACTIC TAUTOLOGIES.",
```

Comment:
Many of the linters have a bit of explanation here about what the error means/how to fix it. Could you write another sentence or two, perhaps with an example and its fix?